src/client/core/utils/array.js

```diff
@@ -66,3 +66,22 @@ export function remove (arr, item) {
     if (index > -1)
         nativeSplice.call(arr, index, 1);
 }
+
+export function equals (arr1, arr2) {
+    if (!arr2)
+        return false;
+
+    if (arr1.length !== arr2.length)
+        return false;
+
+    for (var i = 0, l = arr1.length; i < l; i++) {
+        if (arr1[i] instanceof Array && arr2[i] instanceof Array) {
```

Comment:
it's true, but you didn't answer the question, do we really need this *deep* comparison.

You have two arrays: [parentA1, parentB1, parentC1], [parentA2, parentB2, parentC2]. Why do you need to check `if (arr1[i] instanceof Array && arr2[i] instanceof Array)`?
You just check `arr1.length === arr2.length` and `arr1[n] === arr2[n]`